src/main/java/world/bentobox/bentobox/managers/IslandsManager.java

```diff
@@ -1608,12 +1636,16 @@ public void clearRank(int rank, UUID uniqueId) {
     }
 
     /**
-     * Save the island to the database
+     * Update island data in database
      * 
      * @param island - island
      */
-    public void save(Island island) {
-        handler.saveObjectAsync(island);
+    public static void updateIsland(Island island) {
+        if (handler.objectExists(island.getUniqueId())) {
```

Comment:
The difference between updateIsland and the old save is that this only updates an existing Island. It won't create a new database entry. This prevents async updates from regenerating Island objects that had been deleted.